### PR TITLE
[docs] Add react-native-gradients to ui-programming guides

### PR DESCRIPTION
**Pull request by animeeples**
# Why

Add a Guides page to help answer a common question that new developers have: how to create a gradient in RN.

# How

There were a few ways to do this, but went with the Expo-maintained package.

# Test Plan

Visited http://localhost:3002/ui-programming/react-native-gradients/ and verified that it displayed correctly. Checked links and Snack.

Screen Shot 2021-06-09 at 2 27 38 PM

### CODE REVIEW

File: docs/pages/ui-programming/react-native-gradients.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: Linear gradients in Expo and React Native apps
+sidebar_title: Using linear gradients
+---
+
+import ImageSpotlight from '~/components/plugins/ImageSpotlight';
+import TerminalBlock from '~/components/plugins/TerminalBlock';
+import SnackInline from '~/components/plugins/SnackInline';
+
+To add linear gradients to our Expo and React Native apps, we're going to use `expo-linear-gradient`. We're using this package because it is included in the Expo SDK and also because gradients are not otherwise supported out of the box with React Native.
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-gradients.md:10]:
```suggestion
This page will demonstrate how to style components with linear gradients using the `expo-linear-gradient`. This package provides a simple API and comes pre-installed in the Expo Go and will work in any React Native project, however you are choose to build it.
```

Comment (tcdavis) [reply docs/pages/ui-programming/react-native-gradients.md:10]: A screenshot of the component you've created so folks can immediately see what is being made and confirm its what they're looking for would be great.

File: docs/pages/ui-programming/react-native-gradients.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    color: '#fff',
+  },
+});
+```
+
+</SnackInline>
+
+The example above renders a button like this:
+
+<ImageSpotlight style={{ maxWidth: 276 }} containerStyle={{ marginTop: 0 }} src="/static/images/linear-gradient-expo-rn-example.png" alt="Button with Text over Gradient" />
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-gradients.md:83]: I'd move this towards the top of the page

File: docs/pages/ui-programming/react-native-gradients.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import { LinearGradient } from 'expo-linear-gradient';
+```
+
+Now we can use the `LinearGradient` props `colors`, `start`, `end`, and `locations` to customize and configure our gradient views. 
+
+For more details, check out the [Linear Gradient docs](../versions/latest/sdk/linear-gradient).
+
+## Example
+
+In this example, we are creating a button with text over a gradient.
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-gradients.md:37]: I think a button needs to have an interaction.  I'd either style the background or make an actual button

File: docs/pages/ui-programming/react-native-gradients.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+```js
+import * as React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { LinearGradient } from 'expo-linear-gradient';
+
+export default function App() {
+  return (
+    <View style={styles.container}>
+      <LinearGradient
+        // Button Linear Gradient
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-gradients.md:51]: Not sure this comment makes sense

File: docs/pages/ui-programming/react-native-gradients.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+<!-- prettier-ignore -->
+```js
+import * as React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { LinearGradient } from 'expo-linear-gradient';
+
+export default function App() {
+  return (
+    <View style={styles.container}>
+      <LinearGradient
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-gradients.md:50]: Another example using start and end to change the direction of the gradient would be cool

**Review by tcdavis — CHANGES_REQUESTED** [review]
Thanks for getting this up!

**Comment by amandeepmittal** [comment]
Hi @animeeples, thank you for your contribution!

Unfortunately, I'm closing this PR since we have a working example `expo-linear-gradient` API reference: https://docs.expo.dev/versions/latest/sdk/linear-gradient/. This example also provides a Snack link to quickly run or extend it further and covers the same information. Plus, the contents of this PR are a bit outdated.

**Pull request closed without merging**